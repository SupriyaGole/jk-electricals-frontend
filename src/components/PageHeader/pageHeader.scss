@import "../../styles/variables";

.page-header {
  padding: 0 16px;
  height: 50px;
  display: flex;
  align-items: center;
  background: $green;
  position: relative;
  justify-content: space-between;

  .nav-link:last-child {
    margin-left: 10px;
  }
}

