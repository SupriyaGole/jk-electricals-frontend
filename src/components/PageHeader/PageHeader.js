import React from "react";
import * as PropTypes from "prop-types";
import "./pageHeader.scss";
import {Nav, Navbar} from "react-bootstrap";

const PageHeader = ({title}) => {
  return (
    <header className="page-header">
      <h1 className="page-title">{title}</h1>
      <Navbar bg="dark" variant="dark">
        <Nav className="mr-auto">
          <Nav.Link href="#about">About</Nav.Link>
          <Nav.Link href="#contact-us">Contact Us</Nav.Link>
        </Nav>
      </Navbar>
    </header>
  );
};

PageHeader.propTypes = {
  title: PropTypes.string.isRequired,
};

export default PageHeader;
