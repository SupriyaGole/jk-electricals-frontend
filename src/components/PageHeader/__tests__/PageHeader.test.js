import React from "react";
import PageHeader from "../PageHeader";
import {render} from "@testing-library/react";

describe("Page Header", () => {
  test("should render page header", () => {
    const {baseElement} = render(<PageHeader title="JK Electricals"/>);

    expect(baseElement).toMatchSnapshot();
  });
});
