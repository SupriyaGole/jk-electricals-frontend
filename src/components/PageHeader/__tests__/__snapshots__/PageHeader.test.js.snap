// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Page Header should render page header 1`] = `
<body>
  <div>
    <header
      class="page-header"
    >
      <h1
        class="page-title"
      >
        JK Electricals
      </h1>
      <nav
        class="navbar navbar-expand navbar-dark bg-dark"
      >
        <div
          class="mr-auto navbar-nav"
        >
          <a
            class="nav-link"
            data-rb-event-key="#about"
            href="#about"
          >
            About
          </a>
          <a
            class="nav-link"
            data-rb-event-key="#contact-us"
            href="#contact-us"
          >
            Contact Us
          </a>
        </div>
      </nav>
    </header>
  </div>
</body>
`;
